addons/account/models/product.py

```diff
@@ -55,7 +59,8 @@ def write(self, vals):
     def _get_product_accounts(self):
         return {
             'income': self.property_account_income_id or self.categ_id.property_account_income_categ_id,
-            'expense': self.property_account_expense_id or self.categ_id.property_account_expense_categ_id
+            'expense': self.property_account_expense_id or self.categ_id.property_account_expense_categ_id,
+            'income_refund': self.property_account_income_refund_id or self.categ_id.property_account_income_refund_categ_id
```

Comment:
personally, I think you should set income_refund here to save a bunch of duplications elsewhere (like l1548 of account_invoice and account_invoice_refund. e.g.

`'income_refund': self.property_account_income_refund_id or self.categ_id.property_account_income_refund_categ_id or self.property_account_income_id or self.categ_id.property_account_income_categ_id`